internal/nodes/avi_model_l4_translator.go

```diff
@@ -149,6 +149,11 @@ func (o *AviObjectGraph) ConstructAviL4PolPoolNodes(svcObj *corev1.Service, vsNo
 			if servers := PopulateServersForNPL(poolNode, svcObj.ObjectMeta.Namespace, svcObj.ObjectMeta.Name, false, key); servers != nil {
 				poolNode.Servers = servers
 			}
+		} else if _, ok := svcObj.GetAnnotations()[lib.SkipNodePortAnnotation]; ok {
```

Comment:
Can we change this condition to 
ok || serviceType == lib.NodePort and remove the last else block ?

But the current checks might be simpler .. so this is fine too.

Reply:
Is it the intention to populate the servers from Endpoints as well when serviceType is ClusterIP, and this annotation is present?

Reply:
If someone uses this annotation while the serviceType is clusterIP, the effect is the same as not having the annotation at all.